src/GovUk.Education.ExploreEducationStatistics.Content.Services.Tests/ReleaseFileServiceTests.cs

```diff
@@ -374,11 +372,37 @@ await using (var contentDbContext = InMemoryContentDbContext(contentDbContextId)
             blobStorageService
                 .SetupDownloadToStream(PublicReleaseFiles, releaseFile2.PublicPath(), "Test ancillary blob");
 
+            var subjectIds = releaseFiles
+                .Where(rf => rf.File.SubjectId.HasValue)
+                .Select(rf => rf.File.SubjectId.GetValueOrDefault())
+                .ToList();
+
+            var dataGuidanceFileWriter = new Mock<IDataGuidanceFileWriter>(MockBehavior.Strict);
+
+            dataGuidanceFileWriter
+                .Setup(
+                    s => s.WriteToStream(
+                        It.IsAny<Stream>(),
+                        It.Is<Release>(r => r.Id == release.Id),
+                        It.Is<IEnumerable<Guid>>(
+                            ids => ids.All(id => subjectIds.Contains(id))
+                        )
+                    )
+                )
+                .Returns<Stream, Release, IEnumerable<Guid>?>((stream, _, _) => Task.FromResult(stream))
```

Comment:
Or can you write 

`.ReturnsAsync<Stream, Release, IEnumerable<Guid>?>((stream, _, _) => stream)`

?

Reply:
Nice try but doesn't quite work! This leads to an ambiguous invocation to one of the overloads of `ReturnsAsync`. The real number is this horror show :laughing:

```cs
.ReturnsAsync<Stream, Release, IEnumerable<Guid>?, IDataGuidanceFileWriter, Stream>((stream, _, _) => stream)
```

I'm sure you can understand why I opted for the former.